editor/src/clj/editor/code/view.clj

```diff
@@ -676,6 +676,10 @@
 (g/defnk produce-indent-level-pattern [tab-spaces]
   (data/indent-level-pattern tab-spaces))
 
+(g/defnk produce-document-width [glyph-metrics tab-spaces lines]
+  (let [tab-stops (data/tab-stops glyph-metrics tab-spaces)]
+    (data/max-line-width glyph-metrics tab-stops lines)))
```

Comment:
`document-width` shuold be updated if text metrics changes, so we show horizontal scrollbar in time. We can do it using a listener for change text size event (I did it initially), or using graph and dependencies